packages/block-editor/src/components/block-navigation/list.js

```diff
@@ -44,38 +45,48 @@ export default function BlockNavigationList( {
 	selectedBlockClientId,
 	selectBlock,
 	showNestedBlocks,
+	showMovers,
+	isRootItem = true,
 } ) {
+	const hasMovers = showMovers && blocks.length > 1;
+
 	return (
 		/*
 		 * Disable reason: The `list` ARIA role is redundant but
 		 * Safari+VoiceOver won't announce the list otherwise.
 		 */
 		/* eslint-disable jsx-a11y/no-redundant-roles */
-		<ul className="editor-block-navigation__list block-editor-block-navigation__list" role="list">
+		<ul className="editor-block-navigation__list block-editor-block-navigation__list" role={ isRootItem ? 'tree' : 'group' }>
 			{ map( blocks, ( block ) => {
 				const blockType = getBlockType( block.name );
 				const isSelected = block.clientId === selectedBlockClientId;
+				const blockDisplayName = getBlockDisplayName( blockType, block.attributes );
 
 				return (
-					<li key={ block.clientId }>
-						<div className="editor-block-navigation__item block-editor-block-navigation__item">
+					<li key={ block.clientId } role="treeitem">
+						<div
+							className={ classnames( 'editor-block-navigation__item block-editor-block-navigation__item', {
+								'is-selected': isSelected,
+							} ) }
+						>
 							<Button
-								className={ classnames( 'editor-block-navigation__item-button block-editor-block-navigation__item-button', {
-									'is-selected': isSelected,
-								} ) }
+								className="editor-block-navigation__item-button block-editor-block-navigation__item-button"
 								onClick={ () => selectBlock( block.clientId ) }
 							>
 								<BlockIcon icon={ blockType.icon } showColors />
-								{ getBlockDisplayName( blockType, block.attributes ) }
+								{ blockDisplayName }
 								{ isSelected && <span className="screen-reader-text">{ __( '(selected block)' ) }</span> }
 							</Button>
+							{ hasMovers && ( <BlockMover clientIds={ [ block.clientId ] } /> ) }
```

Comment:
I've switched to using the `isHidden` prop. 

It seems a shame that JS has to be used for this instead of CSS, but in practice it seems like the reason is to replicate `:focus-within`—ensuring the whole mover component continues to be displayed when any of the buttons within the row have focus.